torchinductor/lowering.py

```diff
@@ -251,19 +253,23 @@ def broadcast_symbolic_shapes(a, b):
 
 
 def promote_constants(inputs):
-    if not any(isinstance(x, (int, float)) for x in inputs):
+    if not any(isinstance(x, (sympy.Expr, int, float)) for x in inputs):
```

Comment:
this would need special handling when there are no tensor inputs (@desertfire recently landed a PR with a change here)